frontend/src/components/ingress/Details.tsx

```diff
@@ -42,6 +43,110 @@ export default function IngressDetails() {
     );
   }
 
+  /** this function is used to determine the protocol used in the formatURL for LinkedStringFormat */
+  function getProtocol(item: any, url: any) {
+    const hostList: string[] = item.jsonData.spec?.tls?.map(({ ...hosts }) => `${hosts.hosts}`);
+    const isHttps = hostList.includes(`${url}`);
+
+    return isHttps;
+  }
+
+  /** this function is used to format the url to be used in the Link component */
+  function LinkStringFormat(url: any, item?: any, urlPath?: any) {
+    let urlProtocol;
+
+    if ((url === '*' || url === '' || url === undefined) && !urlPath) {
+      return <Typography>{url}</Typography>;
+    }
+
+    if ((url === '*' || url === '' || url === undefined) && urlPath) {
+      return <Typography>{urlPath}</Typography>;
```

Comment:
Can't this be rewritten as:
```typescript
if (!url || url === '*') {
  return <Typography>{urlPath || url}</Typography>;
}
```
If url is undefined, we probably want to use it as `''` instead, so:
```typescript
if (!url || url === '*') {
  return <Typography>{urlPath || url || ''}</Typography>;
}
```

Reply:
I added notes to the relevant pieces for this to explain why one renders the host field and the other renders the path